src/com/google/enterprise/adaptor/DocumentHandler.java

```diff
@@ -1000,8 +1001,40 @@ private void transform() {
       Map<String, String> params = new HashMap<String, String>();
       params.put("DocId", docId.getUniqueId());
       params.put("Content-Type", contentType);
+      if (null != lastModified) {
+        params.put("Last-Modified-Millis-UTC", "" + lastModified.getTime());
+      }
+      String origDisplayUrl = null;
+      if (null != displayUrl) {
+        origDisplayUrl = "" + displayUrl;
+      }
+      params.put("Display-URL", origDisplayUrl);
+      params.put("Crawl-Once", "" + crawlOnce);
```

Comment:
for lastModified we don't include the key at all if it is null

for displayUrl we explicitly include null if it is null

why not skip the params.put("Display-URL", origDisplayUrl) if displayUrl is null?